src/support/mtx_rw.c

```diff
@@ -148,7 +148,14 @@ __wt_try_readlock(WT_SESSION_IMPL *session, WT_RWLOCK *l)
         return (__wt_set_return(session, EBUSY));
 
     /* We rely on this atomic operation to provide a barrier. */
-    return (__wt_atomic_casv64(&l->u.v, old.u.v, new.u.v) ? 0 : EBUSY);
+    WT_RET(__wt_atomic_casv64(&l->u.v, old.u.v, new.u.v) ? 0 : EBUSY);
+
+#ifdef TSAN_BUILD
+    /* Perform a dummy read to inform TSan this function does in fact have acquire semantics. */
+    (void)__atomic_load_n(&l->tsan_sync, __ATOMIC_ACQUIRE);
```

Comment:
I was a bit worried that this void cast would result in reads being optimised out, but using godbolt and looking at test results on the head of the WT-13154 branch these instructions are emitted in compiled code. I'll do some heavy testing to further support this theory.

Reply:
I think we shouldn't need any additional semantics after a CAS, have you tested without this one?

Reply:
I agree it shouldn't, but local testing seems to require them. If we're adding this reads to some functions I think it's useful to do so consistently, even if only for documentation purposes.